synadm/cli/user.py

```diff
@@ -138,6 +139,55 @@ def deactivate(ctx, helper, user_id, gdpr_erase):
         click.echo("Abort.")
 
 
+@user.command()
+@click.argument("regex", type=str)
+@click.option(
+    "--gdpr-erase", "-e", is_flag=True, default=False,
+    help="""Marks the users as GDPR-erased. This means messages sent by the
+    users will still be visible by anyone that was in the room when these
+    messages were sent, but hidden from other users joining the room
+    afterwards.""", show_default=True)
+@click.option(
+    "--dry-run", "-n", is_flag=True, default=False,
+    help="""Do everything except deactivating users."""
+)
+@click.option(
+    "--batch-size", type=int, default=500, show_default=True,
+    help="""How many users should be requested from the API one at a time.
+    This option has no effect on how many users will be deactivated.
+
+    Increasing this is not necessary in most cases but useful if you have a
+    lot of accounts on your homeserver."""
+)
+@click.pass_obj
+@click.pass_context
+def deactivate_regex(ctx, helper, regex, gdpr_erase, dry_run=False,
+                     batch_size=500):
+    """ Deactivate or GDPR-erase accounts based on regex.
+
+    Does everything normal deactivation does, just for multiple users. The
+    --batch argument may be helpful if you do not want to be prompted to
+    deactivate every single account, and the --dry-run argument is helpful
+    to see which accounts will be deactivated.
+
+    The regex argument takes a string and uses Python's regex matching. It
+    matches based on the full matrix ID."""
```

Comment:
Help texts are precise, not too long and to the point! That is very much appreciated! :heart: 
Because of the last sentecne "....based on the full matrix ID." see below for details but actually we are currently matching even if it's not a full matrix ID (leaving out the @ in the beginning is rather "matching part of the matrix ID). I might be too picky here with the wording so you can as well ignore that and leave it as-is, it's fine, but read on about streamlining below. In what way we need to decide but we should try to "share behaviour" between commands as good as possible.